test/e2e/apimachinery/aggregator.go

```diff
@@ -105,10 +100,13 @@ var _ = SIGDescribe("Aggregator", func() {
 
 })
 
-func cleanTest(client clientset.Interface, aggrclient *aggregatorclient.Clientset, namespace string) {
+func cleanTest(f *framework.Framework, aggrclient *aggregatorclient.Clientset) {
+	client := f.ClientSet
+	namespace := f.Namespace.Name
+
 	// delete the APIService first to avoid causing discovery errors
 	_ = aggrclient.ApiregistrationV1().APIServices().Delete(context.TODO(), "v1alpha1.wardle.example.com", metav1.DeleteOptions{})
-
+	waitAPICoherentResponse(f)
```

Comment:
Do we really want to check for this in `AfterEach`? I was expecting this to be a separate test case.

Reply:
yeah, AfterEach is not the place to assert conditions, if is part of the behavior we want to test it has to be done in the body

Reply:
Was following the same style it already had, as I was not sure  if we wanted to change the test itself. Now I refactored the test so that this function is not needed, let me know what you think. Thanks!